rcgen/src/key_pair.rs

```diff
@@ -166,38 +171,42 @@ impl KeyPair {
 	/// same der key. In that instance, you can use this function to precisely
 	/// specify the `SignatureAlgorithm`.
 	#[cfg(feature = "crypto")]
-	pub fn from_der_and_sign_algo(
-		pkcs8: &[u8],
+	pub fn from_der_and_sign_algo<'a>(
+		der: impl Into<PrivateKeyDer<'a>>,
 		alg: &'static SignatureAlgorithm,
 	) -> Result<Self, Error> {
 		let rng = &SystemRandom::new();
-		let pkcs8_vec = pkcs8.to_vec();
+		let serialized_der = match &der.into() {
```

Comment:
IMO we should take `impl Into<PrivatePkcs8Key<'_>>` instead to make it clear at compile time we only support PKCS#8 keys.